app/models/certificate_service.rb

```diff
@@ -4,4 +4,27 @@ class CertificateService < ActiveRecord::Base
   belongs_to :service, polymorphic: true
 
   validates :certificate, uniqueness: { scope: [:service_id, :service_type] }
+
+  after_create :update_service_id
+  before_destroy :nullify_service_reference
+
+  private
+
+  def service_type_instance
+    service_type.constantize.find(service_id)
+  end
+
+  def update_service_id
+    service_type_instance.tap do |s|
+      s.certificate_service_id = id
+      s.save
+    end
+  end
+
+  def nullify_service_reference
```

Comment:
Is this also necessary? If the service is destroyed, the certificate service will also be destroyed since is dependent.

Reply:
didnt work out of the box so i added this hook

Reply:
Really? Weird... [2]

Reply:
Same for this, we should remove this too.